### ISSUE TITLE
Add actual host architecture telemetry key

### ISSUE THREAD

**Issue by danieljurek** [Azure/azure-dev#2564]
Right now we report `host.arch` as [`runtime.GOARCH`](https://github.com/Azure/azure-dev/blob/572e7f47ee17b3cc10d5eb2380f0ef5e4be54c15/cli/azd/internal/tracing/resource/resource.go#L28)... This indicates the architecture of the compiled binary but not the actual architecture of the host. 

To that end, we should see if we can report the actual architecture of the host (even when running in Rosetta 2). 

This would give us a better idea of the number of people running `azd` on ARM64 machines even if they're running the x86_64 binary.

**Comment by vhvb1989** [comment]
This is interesting. I wouldn't expect folks to run x86/64 on ARM...  How would that be possible?  Would that mean folks manually downloading the binary from gh and executing?  
In most cases, the installer scripts would get the right binary for the user's host, right?  So I wonder if having this extra information would help us with anything?  @danieljurek

**Comment by danieljurek** [comment]
Scenarios where you'd be running x86_64 on ARM64: 

* Windows ARM64: Always running x86_64 on ARM64 unless the user manually downloads and copies the Windows ARM64 build into place (telemetry is useful here) 
* MacOS ARM64: Extremely rarely a user _could_ manually download the x86_64 version and run it with Rosetta 2 installed. This was a more likely scenario back in 2021 and much less likely today
* Linux ARM64: Almost never... I can't envision a scenario where this would be useful because of the amount of work the user would have to do to get Linux ARM64 to run an x86_64 binary when the ARM64 build will be installed as part of the install script

**Comment by vhvb1989** [comment]
> * Windows ARM64: Always running x86_64 on ARM64 unless the user manually downloads and copies the Windows ARM64 build into place (telemetry is useful here)

Does it mean that our installers (winget, choco or pwsh-script) are not detecting the ARM64 from the users and getting that binary?  Are we just defaulting x86_64 for any Windows?

**Comment by danieljurek** [comment]
@vhvb1989, correct. See here for getting Windows ARM64 support to GA: https://github.com/Azure/azure-dev/issues/3968
